src/components/Toolbar.ts

```diff
@@ -74,6 +81,52 @@ const toolbarView = (vnode: Vnode<IToolbar>) => {
           },
           "Import HAR Session"
         ),
+        m(
+          "a.button.is-outlined.is-small.control",
+          {
+            class: ngrokStreamLock === -1 ? "ngrok-play" : "ngrok-stop"
+          },
+          m("hi", {
+            onclick: () => {
+            if (ngrokStreamLock != -1) {
+              console.log('stopping ngrok stream');
+              ngrokStreamLock = -1;
+              return;
+            }
+
+            // this permissions things works but it needs a better way
+            // to test and reset it!
+
+            var API_HOST = 'http://localhost:4040/'; // TODO: make this customisable in settings
+
+            requestPerms(API_HOST).then(() => {
+              let timeoutId = window.setTimeout(function pollStream() {
+                console.log('requesting new data...');
+                let newEntries = fetch(`${API_HOST}api/requests/http`, {
+                  headers: {
+                    'Accept': 'application/json'
+                  }
+                })
+                  .then((response) => response.json())
+                  // and then?
+                  .then(parseNgrokRequests)
+                  // and then and then
+                  .then(({entries}) => vnode.attrs.addRequests(entries));
+
+                  if (ngrokStreamLock === timeoutId) {
```

Comment:
This doesn't wait for a response before setting the next timeout, so it is essentially a `setInterval`. I think this might be better in the `then()` that calls `addRequests()`, or a seperate `.finally()` call in the event of errors.

Otherwise if the response takes close to `ngroxStreamInterval` to respond you might start hammering that endpoint instead of polling politely.